### PR TITLE
[CPM-189] Optimise plan page

### PR DESCRIPTION
**Pull request by daltonlim**
## Changes

- [x] Plan page no longer errors when programme is deleted
- [x] Student page no longer errors when programme is deleted
- [x] Student retrieval on the plan page now gets a single student as opposed to a list of students 
- [x] Backend now has the ability to retrieve single student
- [x] Programme retrieval on the plan page now  retrieves a single program as opposed to a list of programmes
- [x] Programme retrieval on the plan page now  retrieves a single student as opposed to a list of students
- [x] Implement routing to student page on delete

## Checklist

- [x] Have tests been written?
- [x] Is the UI consistent?

## Requests
- [ ] Can somebody check the styling/CSS?

Addresses [CPM-189]

Connects #y

[CPM-189]: https://momentum-agile.atlassian.net/browse/CPM-189

### CODE REVIEW

File: client/src/pages/Plan/Header.js [hunk 1]
```diff
@@ -75,7 +75,7 @@ const Header = ({ name, programme, lastSaveDate, planName, setPlanName, optionsM
 
                         <Flex height="50px" backgroundColor={c.lightBlue} borderRadius="10px" align="center" justify="center" p={5} mt={2}>
                             <Text textAlign="center" color={c.white} fontSize="xl">
-                                {programme}
+                                {programme ? programme : "Program could not be found"}
```
Comment (FocalChord) [inline client/src/pages/Plan/Header.js:78]: `programme || "Program could not be found"`

File: client/src/pages/Plan/RequirementsList.js [hunk 2]
```diff
@@ -5,7 +5,8 @@ import { ProgrammeRequirementsItem } from "../../components";
 import usePlan from "./usePlan";
 
 const RequirementsList = ({ programme }) => {
-    const regulations = programme.regulations;
+    console.log(programme);
```
Comment (FocalChord) [inline client/src/pages/Plan/RequirementsList.js:8]: Remove

File: client/src/pages/Students/editables/PlansPane.js [hunk 3]
```diff
@@ -149,10 +149,10 @@ const PlansTableItem = ({ plan: { _id, name, numYears, startYear }, programme })
                         color={programme ? c.lightBlue : c.red}
                         rounded="10px"
                         onClick={(e) => {
-                            e.stopPropagation();
-                            window.open(`/programmes/${programme._id}`, "_blank");
+                            programme && e.stopPropagation();
+                            programme && window.open(`/programmes/${programme._id}`, "_blank");
                         }}
```
Comment (FocalChord) [inline client/src/pages/Students/editables/PlansPane.js:154]: Why not chain all of them into one line xd

programme && e.stopPropogation() && window.open(`/programmes/${programme._id}`, "_blank");

Comment (daltonlim) [reply client/src/pages/Students/editables/PlansPane.js:154]: chaining doesnt seem to work :(

File: client/src/pages/Plan/RequirementsList.js (hunk 2 above)
Comment (daltonlim) [reply client/src/pages/Plan/RequirementsList.js]: my bad. resolved now

File: client/src/pages/Students/editables/PlansPane.js (hunk 3 above)
Comment (FocalChord) [reply client/src/pages/Students/editables/PlansPane.js:154]: rip

File: client/src/pages/Plan/Header.js [hunk 4]
```diff
@@ -75,7 +75,7 @@ const Header = ({ name, programme, lastSaveDate, planName, setPlanName, optionsM
 
                         <Flex height="50px" backgroundColor={c.lightBlue} borderRadius="10px" align="center" justify="center" p={5} mt={2}>
                             <Text textAlign="center" color={c.white} fontSize="xl">
-                                {programme ? programme : "Program could not be found"}
+                                {programme || "Program could not be found"}
```
Comment (FocalChord) [inline client/src/pages/Plan/Header.js:78]: Should we make this

"Programme could not be found" instead of "Program could not be found"

Comment (daltonlim) [reply client/src/pages/Plan/Header.js]: sgtm

File: client/src/pages/Plan/RequirementsList.js [hunk 5]
```diff
@@ -5,7 +5,7 @@ import { ProgrammeRequirementsItem } from "../../components";
 import usePlan from "./usePlan";
 
 const RequirementsList = ({ programme }) => {
-    const regulations = programme.regulations;
+    const regulations = (programme && programme.regulations) || [];
```
Comment (FocalChord) [inline client/src/pages/Plan/RequirementsList.js:8]: I think this line is the same as 

`const regulations = programme ? programme.regulations : []`

**Review by FocalChord — APPROVED** [review]
lgtm

**Review by anian100 — APPROVED** [review]
lgtm 👍

**Pull request merged**